modules/fun.js

```diff
@@ -0,0 +1,450 @@
+    case `Hard`:
+      size = 14;
+      mineCount = 60;
+      break;
+    case `Medium`:
+      size = 10;
+      mineCount = 30;
+      break;
+    default:
+    case `Easy`:
+      size = 5;
+      mineCount = 5;
+      break;
+  }
+  // generate the minefield
+  let field = "";
+  // Getting all possible board spaces
+  const possibleSpaces = Array.from({ length: size * size }, (v, k) => k);
+  // Remove 4 corners, corners can't be mines
+  possibleSpaces.splice((size * size) - 1, 1);
+  possibleSpaces.splice((size - 1) * size, 1);
+  possibleSpaces.splice(size - 1, 1);
+  possibleSpaces.splice(0, 1);
+  // Finding out where the mines will be
+  const mineSpaces = [];
+  for (let i = 0; i < mineCount; i++) {
+    const random = Math.floor(Math.random() * possibleSpaces.length);
+    mineSpaces.push(possibleSpaces[random]);
+    possibleSpaces.splice(random, 1);
+  }
+
+  function getMineCount(x, y) {
+    let count = 0;
+    for (let i = -1; i <= 1; i++) {
+      if ((x + i) < 0 || (x + i) >= size) continue;
+      for (let j = -1; j <= 1; j++) {
+        if ((y + j) < 0 || (y + j) >= size) continue;
+        if (mineSpaces.includes((y + j) * size + x + i)) count++;
+      }
+    }
+
+    return count;
+  }
+
+  // Creating the final board
+  /** @type {number[][]} */
+  const board = [];
+  for (let x = 0; x < size; x++) {
+    board.push([]);
+    for (let y = 0; y < size; y++) {
+      if (mineSpaces.includes(x + (y * size))) {
+        board[x].push(9);
+        continue;
+      }
+      board[x].push(getMineCount(x, y));
+    }
+  }
+  const output = board.map(row => row.map(num => `||${num == 9 ? mineSweeperEmojis[`bomb`] : mineSweeperEmojis[num]}||`).join(``)).join(`\n`);
+  field = (`**Mines: ${mineCount}** (Tip: Corners are never mines)\n${output}`);
+  // we need to split it up because only 99 emoji per message limit for some reason.
+  let degradingField = field;
+  function countEmoji(text) {
+    const emojiRegex = new RegExp(`(${Object.values(mineSweeperEmojis).map(emoji => emoji.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')).join(`|`)})`, 'g');
+    const emoji = text.match(emojiRegex);
+    return emoji?.length || 0;
+  }
+  if (countEmoji(field) <= 99) {
+    return int.editReply(field); // No splitting needed
+  }
+  while (countEmoji(degradingField) > 99) {
+    let segment = ``;
+    while (countEmoji(segment + degradingField.substring(0, degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length)) <= 99) {
+      segment += degradingField.substring(0, (degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+      degradingField = degradingField.substring((degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+    }
+    if (segment + degradingField == field) {
+      await int.editReply(segment);
+    } else {
+      await int.channel.send(segment);
+    }
+  }
+  return int.channel.send(degradingField);
```

Comment:
soooo it might be a character limit instead of an emoji limit? might be wrong tho so double check (also you might have to replace the emojis at the start of the file with the :emoji: version)

```suggestion
  rows.unshift(`**Mines: ${mineCount}** (Tip: Corners are never mines)`);
  const msgs = []
  let msg = ""
  rows.forEach((str) => {
    if (msg.length + str.length > 3900) {
      msgs.push(msg)
      msg = ""
    }
    msg += `${str}\n`;
  })
  await int.editReply(msgs[0])
  if (msgs[1] int.followUp(msgs[1])
```

Reply:
I did specific testing. just the minefeild without a header cuts out at the same point, plus its 99 that it cuts off at, which wouldn't make sense with a non round number... unless each emoji counts as 20 characters, thus making 100 emoji = 2000 characters which I know is the limit for bots because I got an error with it somewhere, in fact it may have been the error...

Reply:
after a bit of expirimenting, it does in fact seem to be that each emoji is probably 20 characters. handicorn seems to be 32 characters at least from the user POV...

Reply:
I dont know, I just know that this at least works for all 3 sizes.

Reply:
in case this is a problem with other future functions, perhaps this should be a utils function once I figure it out?

Reply:
after extensive pondering and testing, it is a tag pair limit. I presume discord most likely converts it to HTML or somesuch and each emoji has a tag pair and each spoiler has a tag pair, and it limits how many pairs to 199.
testing done here: https://discord.com/channels/1207041599608061962/1207044664411291719/1281456134061953035

Reply:
I don't know how to accurately handle that, or if this is a bug on discord's end...

Reply:
I've reported it as a bug on discords end, as it definitely seems to be with how "copy text" still copies the whole thing... whether they document it as a limit or fix it is yet to be known, but till then I think this calls for some strange handling.

Reply:
Ive got some temporary stuff set up. we should probably keep an issue or something open in case there is ever a solution.

Reply:
yknow what, scratch that. it works well enough as it is. this can last until we come back to update or change it again.